### PR TITLE
Resolve control node identity via system.local before initial metadata refresh

### PR DESCRIPTION
**Pull request by dkropachev**
## Summary

Refactors how contact points transition into metadata nodes:

- **ControlConnection resolves contact points eagerly**: After opening a channel to a contact point, the control connection queries `system.local` via `getChannelNodeInfo()` and registers the node into metadata (`MetadataManager.registerNode()`) before `InitialNodeListRefresh` runs.
- **InitialNodeListRefresh simplified**: No longer receives contact points. Creates new nodes for all discovered hosts, reusing any pre-registered node (matched by hostId) to preserve connection state.
- **New `MetadataManager.registerNode()`**: Atomically creates and inserts a metadata node from `NodeInfo` on the admin executor, preventing races with concurrent metadata refreshes.
- **Control node tracking by hostId**: New `controlNode()` accessor and `isControlNode()` method match by hostId instead of endpoint comparison, fixing false matches behind NLB/proxy where multiple nodes share the same endpoint.
- **Local DC inference from control connection**: LBP helpers (`OptionalLocalDcHelper`, `MandatoryLocalDcHelper`, `InferringLocalDcHelper`) infer the local DC from the control connection node instead of contact points, removing `wasImplicitContactPoint()` and contact-point DC checks.
- **Contact points without metrics**: `DefaultNode.newContactPoint()` creates nodes with `NoopNodeMetricUpdater` since contact points are temporary objects replaced by real metadata nodes.
- **TopologyMonitor API**: `getChannelEndpoint()` renamed to `getChannelNodeInfo()`, returning full `NodeInfo` instead of just `EndPoint`.
- **Edge case handling**: Channel closed during resolve, control node removed from metadata on reconnect, empty proxy targets in `RoundRobinProxy`, NlbSimulator cleanup when all nodes removed.

## Test plan

- [x] Existing unit tests updated and pass: `ControlConnectionTest`, `InitialNodeListRefreshTest`, `MetadataManagerTest`, LBP init/distance tests
- [x] New unit tests: control node lifecycle (resolve, clear on close, restore after reconnect), `registerNode` idempotency, contact-point-to-metadata-node resolution
- [x] Integration tests updated: `ConnectIT` (DC inference from control connection, LBP init failure cleanup), `NodeStateIT` (event expectations for new node lifecycle)
- [x] CI/CD pipeline passes

### CODE REVIEW

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java [hunk 1]
```diff
@@ -606,13 +674,29 @@ private void reconnectNow() {
...
+      Node current = currentControlNode;
+      Node pending = pendingControlNode;
+      if (current != null
+          && eventNode.getHostId() != null
+          && eventNode.getHostId().equals(current.getHostId())) {
+        return true;
+      }
+      if (current == null
+          && pending != null
+          && Objects.equals(eventNode.getEndPoint(), pending.getEndPoint())) {
```
Comment (nikagra) [inline core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java]: So, do we still have comparison as a fallback?

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java [hunk 2]
```diff
@@ -446,38 +461,54 @@ private void connect(
...
+                      currentControlNode = null;
+                      if (previousChannel != null && previousChannel != channel) {
                         LOG.debug(
                             "[{}] Forcefully closing previous channel {}",
                             logPrefix,
                             previousChannel);
                         previousChannel.forceClose();
                       }
-                      resolveChannelNodeInfo(channel)
+                      resolveChannelNodeIfNeeded(channel, (DefaultNode) node)
```
Comment (nikagra) [inline core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java:471]: I think adding a guard here may be a good idea

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java [hunk 3]
```diff
@@ -606,13 +674,29 @@ private void reconnectNow() {
       }
     }
 
+    private boolean isControlNode(Node eventNode) {
+      Node current = currentControlNode;
+      Node pending = pendingControlNode;
```
Comment (nikagra) [inline core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java]: Is reading two volatile fields safe here?

Comment (dkropachev) [reply core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java]: semi - safe, in theory it is classic thread-safety problem, in practice we will never hit it.

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java (hunk 1 above)
Comment (dkropachev) [reply core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java]: correct

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java (hunk 2 above)
Comment (dkropachev) [reply core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java:471]: nope, Node is always a DefaultNodeis in this repo casting like this all around the code

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java (hunk 3 above)
Comment (dkropachev) [reply core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java]: ok, let me do immutable holder for these guys, to make it bulletproof

File: core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java [hunk 4]
```diff
@@ -446,38 +460,53 @@ private void connect(
...
+                              (resolvedNode, fetchError) -> {
                                 if (fetchError != null) {
+                                  controlNodeState = ControlNodeState.NONE;
                                   LOG.debug(
                                       "[{}] Failed to resolve control node endpoint from {}, "
                                           + "trying next node",
                                       logPrefix,
                                       node,
                                       fetchError);
+                                  ControlConnection.this.channel = null;
```
Comment (nikagra) [inline core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java:485]:
```suggestion
                                  // Null out before forceClose() so that onChannelClosed() does not
                                  // start a redundant reconnection on top of the connect() retry below.
                                  ControlConnection.this.channel = null;
```

**Review by nikagra — APPROVED** [review]
One clarifying comment suggestion, 👍  otherwise
